### PR TITLE
rle parser method

### PR DESCRIPTION
**Pull request by SheyReiff**

### CODE REVIEW

File: src/main/java/reiff/gameoflife/GameOfLife.java [hunk 1]
```diff
@@ -41,6 +41,70 @@ public String toString() {
...
     }
 
+    public void loadRleFromString(String rle) {
+
+        if (rle == null || rle.isEmpty()) {
+            this.width = 5;
+            this.height = 3;
+            this.field = new int[height][width];
+            return;
+        }
```
Comment (andrewoid) [inline src/main/java/reiff/gameoflife/GameOfLife.java]: If it's null or empty, don't do anything.

File: src/test/java/reiff/gameoflife/GameOfLifeTest.java [hunk 2]
```diff
@@ -45,4 +47,71 @@ public void nextGen() {
...
+
+        // when
+        game.loadRleFromString(rle);
+
+        // then
+        assertEquals("01000\n00100\n11100\n", game.toString());
+    }
+
+    @Test
+    public void testRleParsingSingleRow() {
```
Comment (andrewoid) [inline src/test/java/reiff/gameoflife/GameOfLifeTest.java]: don't use the word "test" in your test methods.

File: src/test/java/reiff/gameoflife/GameOfLifeTest.java [hunk 3]
```diff
@@ -45,4 +47,71 @@ public void nextGen() {
...
+
+        // When
+        game.loadRleFromString(rle);
+
+        // Then
+        assertEquals("01000\n", game.toString());
+    }
+
+    @Test
+    public void testEmptyRleInput() {
```
Comment (andrewoid) [inline src/test/java/reiff/gameoflife/GameOfLifeTest.java]: same as above

File: src/test/java/reiff/gameoflife/GameOfLifeTest.java [hunk 4]
```diff
@@ -45,4 +47,71 @@ public void nextGen() {
...
+
+        // When
+        game.loadRleFromString(rle);
+
+        // Then
+        assertEquals("00000\n00000\n00000\n", game.toString());
+    }
+
+    @Test
+    public void testOnlyLiveCells() {
```
Comment (andrewoid) [inline src/test/java/reiff/gameoflife/GameOfLifeTest.java]: same as above

File: src/test/java/reiff/gameoflife/GameOfLifeTest.java [hunk 5]
```diff
@@ -45,4 +47,71 @@ public void nextGen() {
...
+
+        // When
+        game.loadRleFromString(rle);
+
+        // Then
+        assertEquals("11111\n", game.toString());
+    }
+
+    @Test
+    public void testLongDeadCellRuns() {
```
Comment (andrewoid) [inline src/test/java/reiff/gameoflife/GameOfLifeTest.java]: same as above

File: src/main/java/reiff/gameoflife/GameOfLife.java (hunk 1 above)
Comment (SheyReiff) [reply src/main/java/reiff/gameoflife/GameOfLife.java]: Do you mean that I should delete this condition to begin with or just do return if it null or empty?

Comment (andrewoid) [reply src/main/java/reiff/gameoflife/GameOfLife.java]: You can delete it.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**